Add `contribute/framework` doc

kylegach opened the pull request:
## What I did

Add a page at the `/docs/<framework>/contribute/framework` route explaining how to contribute a Storybook framework

kylegach committed:
Add `contribute/framework` doc

shilman reviewed (approved):
This is great!

Historically, people have wanted to contribute frameworks directly to the monorepo--I wonder if this doc should somehow preempt that.

I also wonder about a step on distributing the framework, e.g. tweet and mention `@storybookjs`. Or if the pattern described here will automatically get picked up by the integrations catalog, mention that?

kylegach committed:
Address feedback

- Add bits about announcing and path to "official" status

jonniebigodes commented on docs/contribute/framework.md:
```diff
@@ -0,0 +1,145 @@
+
+The library or libraries your framework supports may have different major versions available. Carefully consider which versions of each library your framework will support. You will need to account for the changes within those different versions or split your framework into different versions/packages itself to support each library version. To help ongoing maintenance, please consider adding inte ...
+
+### 3. Write the README
+
+Before writing any code, we advise you to write up a helpful README. It should, at minimum, contain installation instructions and a list of available features. Use the [README for `@storybook/nextjs`](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/README.md) as a template. By writing the documentation first, it becomes a guide for your other work.
```
@kylegach we can probably rename this  "Write documentation". What do you think?

jonniebigodes commented on docs/contribute/framework.md:
```diff
@@ -0,0 +1,145 @@
+
+Once it's fully tested and released, please let us know about your framework by either announcing it in the `#showcase` channel of the [Storybook Discord](https://discord.gg/storybook) or tweeting it and mentioning `@storybookjs`. It's our hope that well-made community frameworks can eventually move into the Storybook codebase and be considered "official" support.
+
+## Thank you
+
+We deeply appreciate your help in making Storybook work for more developers and the libraries in which they build. If you have any suggestions to improve this document, please use the “Edit” link below.
```
@kylegach, if this is going to be added to the documentation, we don't need this. Instead, we could replace this with something linking to the other parts of the documentation of how to contribute, something like:
```md
---
Learn about other ways you can contribute to Storybook:
- [Documentation](./documentation.md) 
- [Examples](./new-snippets.md) 
- [Code](./code.md)
```

If you're ok with this, we can adjust the other contribution pages

jonniebigodes reviewed (changes requested):
@kylegach, this looks great. Just left a couple of items for you to look at. Let me know if you're ok with it, and we'll go from there also while we're at it. It seems that you left out the toc.js entry so we can hook this up to the documentation. Was this intentional or not?

kylegach committed:
Address feedback

- Add link to TOC
- Change 'Write the README' heading to 'Write the documentation"
- Remove 'Thank you' section
- Add 'Other ways to contribute' section
    - Update/add corresponding section on other relevant pages

jonniebigodes reviewed (approved):
Thanks for taking care of the feedback so promptly @kylegach! Appreciate it. All good on my end! Merge when able
